validator/client/src/test/java/tech/pegasys/teku/validator/client/loader/KeystoreLockerTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client.loader;
+
+import static java.nio.file.StandardOpenOption.CREATE_NEW;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static tech.pegasys.teku.validator.client.loader.KeystoreLocker.deleteIfStaleLockfileExists;
+import static tech.pegasys.teku.validator.client.loader.KeystoreLocker.longPidToNativeByteArray;
+
+import com.google.common.io.Resources;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.util.config.InvalidConfigurationException;
+
+public class KeystoreLockerTest {
+
+  private KeystoreLocker keystoreLocker = new KeystoreLocker();
+
+  @Test
+  void shouldLockKeystoreFileAndFailWhenTryingCreateLockForLockedFile() throws Exception {
+    final Path keystoreFile = Path.of(Resources.getResource("scryptTestVector.json").toURI());
```

Comment:
Just realised we should be copying this to a temp dir rather than assuming we can write to a file beside the resource - theoretically the tests could be running from inside JAR files and we don't want tests writing to source directories regardless.

Also, the keystore doesn't actually have to exist or be valid - it's just a path we're passing to `KeystoreLocker`.  So just create a temp dir and then point to a non-existent keystore file in that dir (ie `<tempdir>/keystore.json` but no need for `keystore.json` to actually exist.

Reply:
Done.